src/Illuminate/Database/Eloquent/Concerns/HasAttributes.php

```diff
@@ -1594,32 +1624,46 @@ public function originalIsEquivalent($key)
             return false;
         }
 
-        $attribute = Arr::get($this->attributes, $key);
-        $original = Arr::get($this->original, $key);
+        return $this->attributeValueIsEquivalent(
+            $key,
+            Arr::get($this->attributes, $key),
+            Arr::get($this->original, $key)
+        );
+    }
 
-        if ($attribute === $original) {
+    /**
+     * Determine if two values of a given attribute are equivalent.
+     *
+     * @param  string  $key
+     * @param  array  $newValues
+     * @param  array  $oldValues
+     * @return bool
+     */
+    public function attributeValueIsEquivalent($key, $value1, $value2)
+    {
+        if ($value1 === $value2) {
             return true;
-        } elseif (is_null($attribute)) {
+        } elseif (is_null($value1)) {
             return false;
         } elseif ($this->isDateAttribute($key)) {
-            return $this->fromDateTime($attribute) ===
-                   $this->fromDateTime($original);
+            return $this->fromDateTime($value1) ===
+                   $this->fromDateTime($value2);
```

Comment:
Can we keep the "attribute/original" terminology here? "value1" & "value2" is a bit vague.

Reply:
It doesn't mean the same thing in the new context.